File: src/Entity/Interaction.php
<?php

namespace App\Entity;

use App\Repository\InteractionRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * @ORM\Entity(repositoryClass=InteractionRepository::class)
 */
class Interaction
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("interaction")
     */
    private $idi;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups("interaction")
     */
    private $type;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     * @Groups("interaction")
     */
    private $descrp;



    /**
     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="Interactions")
     * @ORM\JoinColumn(nullable=false)
     * @Groups("interaction")
     */
    private $user;

    /**
     * @ORM\ManyToOne(targetEntity=Article::class, inversedBy="Interaction")
     * @ORM\JoinColumn(nullable=false)
     */
    private $article;



    public function getIdi(): ?int
    {
        return $this->idi;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getDescrp(): ?string
    {
        return $this->descrp;
    }

    public function setDescrp(?string $descrp): self
    {
        $this->descrp = $descrp;

        return $this;
    }

    public function getUser(): ?user
    {
        return $this->user;
    }

    public function setUser(?user $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getArticle(): ?Article
    {
        return $this->article;
    }

    public function setArticle(?Article $article): self
    {
        $this->article = $article;

        return $this;
    }

}

File: src/com/esprit/entities/Interaction.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.esprit.entities;

/**
 *
 * @author ksaay
 */
public class Interaction {
    int idi;
    String type;
    String descrp;
    User user;
    Article article;

    @Override
    public String toString() {
        return "Interaction{" + "idi=" + idi + ", type=" + type + ", descrp=" + descrp + ", user=" + user + ", article=" + article + '}';
    }

    public void setIdi(int idi) {
        this.idi = idi;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setDescrp(String descrp) {
        this.descrp = descrp;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public void setArticle(Article article) {
        this.article = article;
    }

    public int getIdi() {
        return idi;
    }

    public String getType() {
        return type;
    }

    public String getDescrp() {
        return descrp;
    }

    public User getUser() {
        return user;
    }

    public Article getArticle() {
        return article;
    }

    public Interaction(String type, String descrp, User user, Article article) {
        this.type = type;
        this.descrp = descrp;
        this.user = user;
        this.article = article;
    }

    public Interaction(int idi, String type, String descrp, User user, Article article) {
        this.idi = idi;
        this.type = type;
        this.descrp = descrp;
        this.user = user;
        this.article = article;
    }
}
